public/js/graph/react_graph.js

```diff
@@ -438,6 +440,34 @@ var Graph = React.createClass({
         });
     },
 
+    onButtonPress: function(zoomOrPanFunction, direction) {
+        this.setState({mouseDown: setInterval(() => zoomOrPanFunction(direction), 100)});
```

Comment:
Sort of. The problem is `setState` is asynchronous, so even if `setState` is called at each interval, that's not necessarily going to change the state right away, nor is it going to re-render the svg right away (that's why the zooming is a little choppy, but it's less noticeable when there's more of a delay). I'm not really sure what your solution is, since it seems like the existing code already puts the return value of `setInterval` as the `mouseDown` state.

However, I did some research and perhaps I was wrong about this being a "hack" at all. E.g., this is basically the approach used in https://facebook.github.io/react/docs/reusable-components.html#mixins. Maybe the only issue is when the period is too short. So this might not be worth worrying about.

As for the arrow/scroll events... don't worry about that yet (even though it is a little annoying). I think the best thing to do in that case is use an external library for help. You might be interested in checking out https://lodash.com/docs#throttle